corehq/apps/locations/models.py

```diff
@@ -680,7 +680,11 @@ def save(self, *args, **kwargs):
         kwargs['sync_to_sql'] = False  # only sync here
         super(Location, self).save(*args, **kwargs)
         if sync_to_sql:
-            self._migration_do_sync()
+            try:
+                self._migration_do_sync()
+            except Exception:
+                self.delete()  # roll back on failure to sync
```

Comment:
i'm not sure we want to do this - suppose postgres goes down, then any existing location we try to save we'd end up deleting right?